### PR TITLE
Feature/snowflake

### PR DESCRIPTION
**Pull request by eogilvy12**
### Snowflake compatibility

* update package to use macros
* rename sensitive field names (like `time` and `identity`) in all models with if statements 
* update project.yml to reference the project

Questions: 
* should we update `events` to be `pageviews`? This naming convention was confusing for me
* i also don't have a redshift client to test this on - can you ensure this works? I don't think it should have any breaking changes

### CODE REVIEW

File: macros/base/heap_sessions.sql [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+),
+
+cleaned as (
+
+    select
+
+        user_id,
+        session_id,
+        {% if target.type == 'redshift' %}
```
Comment (jthandy) [inline macros/base/heap_sessions.sql]: see above.

File: macros/base/heap_events.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+{% endmacro %}
+
+{% macro default__heap_events() %}
+
+select 
+    
+    *,
+    
+    {% if target.type == 'redshift' %}
```
Comment (jthandy) [inline macros/base/heap_events.sql]: this is the exact kind of check that we don't want to do--it's the whole point of the adapter macro pattern. the problem with this if/then logic is that it doesn't extend well as we continue to add other adapters, so we want to stay away from doing it altogether; you'll notice that our packages never contain logic like this. the way we'd solve this is to write a short adapter macro just called `time_field_name()` or something like that, and then the implementations for the different adapters would be different. I'm happy to discuss this with you more, I get that you probably haven't had to do exactly this thing before.

File: macros/base/heap_users.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        between unbounded preceding and unbounded following"
+%}
+
+--this is only done because heap has a bug where it inserts duplicate records for users.
+
+select distinct
+
+    user_id,
+    last_value(
+        {% if target.type == 'redshift' %}
```
Comment (jthandy) [inline macros/base/heap_users.sql]: see above.

File: macros/base/heap_events.sql (hunk 2 above)
Comment (eogilvy12) [reply macros/base/heap_events.sql]: @jthandy would love to discuss this as soon as possible as I have a delivery tomorrow afternoon and need to move this forward. this was my attempt to not replicate code as there were such small alterations needed to be compatible.

**Comment by eogilvy12** [comment]
@jthandy ready to take a look at!

**Pull request merged**